usr/bin/whonix-gateway-firewall

```diff
@@ -640,6 +655,9 @@ ipv4_output() {
    ## Prevent connections to Tor SocksPorts.
    ## https://phabricator.whonix.org/T533#11025
    if [ "$firewall_mode" = "timesync-fail-closed" ]; then
+      if [ "${info_enabled}" = "1" ]; then
+        output_cmd "INFO: firewall_mode=${firewall_mode}, closing Internal TCP port(s): ${INTERNAL_OPEN_PORTS}, except 9108 for sdwdate"
```

Comment:
Also cool. Just perhaps we can improve the wording? We're not really closing any ports here. We're not opening them. Nitpicky technical difference? Not sure. Otherwise someone could ask "how to close all ports except 9108"?

Perhaps

> keeping internal TCP ports closed: ${INTERNAL_OPEN_PORTS} (except 9108 for sdwdate)

?

Reply:
Or...

> not opening internal TCP ports: ${INTERNAL_OPEN_PORTS} (except 9108 for sdwdate)

Reply:
https://github.com/Whonix/whonix-firewall/pull/1/commits/5ec6e1fac57c5dd6d5f6a80170c61476e95141b3